app/models/concerns/archivable.rb

```diff
@@ -0,0 +1,133 @@
+require 'csv'
+
+# Methods pertaining to patient export
+module Archivable
+  extend ActiveSupport::Concern
+
+  def has_alt_contact?
+    other_contact.present? || other_phone.present? || other_contact_relationship.present?
+  end
+
+  def age_range
+    case age
+    when nil, ''
+      nil
+    when 1..17
+      'Under 18'
+    when 18..24
+      '18-24'
+    when 25..34
+      '25-34'
+    when 35..44
+      '35-44'
+    when 45..54
+      '45-54'
+    when 55..100
+      '55+'
+    else
+      'Bad value'
+    end
+  end
+
+  class_methods do
+    def unarchived
+      Patient.where(archived: :false)
+    end
+
+    def archive
+      Patient.ready_to_archive do |patient|
+        patient.attributes.keys do |field|
+          # lookup action for key in hash
+          # if hash key doesn't exist, delete it and log a warning to update the hash!
+        end
+        # DUMMY
+        # TODO erase fields
+        # TODO scramble un-erasable fields
+        patient.archived = true
```

Comment:
What are your thoughts about having this method do something like 

```
Patient.ready_to_archive do |patient|
  patient.archive!
end
```

and have the instance archive-with-a-bang method do the heavy lifting/transformation? I think that'll make it way easier to test, and also let us archive individual patients one at a time. (Apologies if this is on your radar and just hasn't gotten to that point yet!)